napari/layers/vectors/vectors.py

```diff
@@ -653,10 +653,14 @@ def _slice_data(
             values, based on how far from the current slice they originate.
         """
         not_disp = list(self._dims_not_displayed)
-        indices = np.array(dims_indices)
+        # We want a numpy array so we can use fancy indexing with the non-displayed
+        # indices, but as dims_indices can (and often/always does) contain slice
+        # objects, the array has dtype=object which is then very slow for the
+        # arithmetic below.
```

Comment:
This is identical to the comment in `Points._slice_data`, except the bit about `Points._round_index` which is missing for `Vectors`, so slice indices are currently rounded. That means that casting to `float` here is in general lossy, but only for really large integers.

We might consider using the following form instead to allow numpy to decide what dtype to use (i.e. `int` vs. `float`): https://github.com/napari/napari/pull/4648/files#r893051191

Alternatively, maybe `Vectors` layers shouldn't be casting slice indices to ints for the same reason that `Points` layers shouldn't.